crates/cxx-qt-gen/src/generator/rust/constructor.rs

```diff
@@ -0,0 +1,310 @@
+            &constructor.items.new_arguments,
+            format_ident!("new_arguments"),
+        );
+        let init_initialize_arguments = extract_arguments_from_tuple(
+            &constructor.items.initialize_arguments,
+            format_ident!("initialize_arguments"),
+        );
+        let init_base_arguments = extract_arguments_from_tuple(
+            &constructor.items.base_arguments,
+            format_ident!("base_arguments"),
+        );
+
+        let extract_new_arguments = extract_arguments_from_struct(
+            &constructor.items.new_arguments,
+            format_ident!("new_arguments"),
+        );
+
+        let extract_initialize_arguments = extract_arguments_from_struct(
+            &constructor.items.initialize_arguments,
+            format_ident!("initialize_arguments"),
+        );
+
+        result.cxx_mod_contents.append(&mut vec![
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #arguments_cxx]
+                #[doc(hidden)]
+                struct #arguments_rust {
+                    baseArguments: #base_arguments_rust,
+                    newArguments: #new_arguments_rust,
+                    initializeArguments: #initialize_arguments_rust,
+                }
+            },
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #base_arguments_cxx]
+                #[doc(hidden)]
+                struct #base_arguments_rust {
+                    #(#base_argument_members,)*
+                    notEmpty: i8 // Make sure there's always at least one struct member, as CXX
+                                 // doesn't support empty shared structs.
+                }
+            },
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #new_arguments_cxx]
+                #[doc(hidden)]
+                struct #new_arguments_rust {
+                    #(#new_argument_members,)*
+                    notEmpty: i8 // Make sure there's always at least one struct member, as CXX
+                                 // doesn't support empty shared structs.
+                }
+            },
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #initialize_arguments_cxx]
+                #[doc(hidden)]
+                struct #initialize_arguments_rust {
+                    #(#initialize_argument_members,)*
+                    notEmpty: i8 // Make sure there's always at least one struct member, as CXX
+                                 // doesn't support empty shared structs.
+                }
+            },
+            parse_quote! {
+                extern "Rust" {
+                    #[namespace = #namespace_internals]
+                    #[cxx_name = #route_arguemnts_cxx]
+                    // This function needs to marked unsafe, as some arguments may be pointers.
```

Comment:
Can you not use the `is_unsafe_cxx_type` to determine this, if it only happens sometimes ?

If it always happens then rephrase to state there'll always be pointers

Reply:
Turns out I can.
Done.